lib/client-assets.php

```diff
@@ -1216,6 +1216,9 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	if ( $post_autosave ) {
 		$editor_settings['autosave'] = array(
 			'editLink' => add_query_arg( 'gutenberg', true, get_edit_post_link( $post_autosave->ID ) ),
+			'title'    => $post_autosave->post_title,
```

Comment:
This does not take the shape nor apply the same filtering as is done in the REST revisions controller which is otherwise used by the application, and probably should.

https://github.com/WordPress/WordPress/blob/4c2fcf36a3193a7e63a82e5ff459f2587c69a859/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php#L386-L407

Maybe `get_autosave_newer_than_post_save` should be returning a REST entity? Seems like we have two shapes of an autosave between how we're bootstrapping vs. otherwise using in the editor, and I don't see why it ought to be that way.

Reply:
Would it be best to extract the building of the autosave object from the rest api code? Maybe relocate it to the revisions.php file?

Reply:
The [`gutenberg_preload_api_request` function](https://github.com/WordPress/gutenberg/blob/b06a33569a2b860110cb8d3f113ccf6c2b36e492/lib/client-assets.php#L566-L618) could probably server as prior art here.